src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -107,7 +107,7 @@ public class JUnitResultArchiver extends Recorder implements SimpleBuildStep, JU
      */
     private boolean skipMarkingBuildUnstable;
 
-    private boolean parseOldReports;
+    private boolean parseOldReports = true;
```

Comment:
Beware that this will not take effect when loading existing non-Pipeline jobs from disk: the field will be missing from `config.xml` and so will default to `false`. Meaning the pre-#348 behavior. Whereas for Pipeline jobs the `JUnitResultStep` object is created via constructor every time, so it will default to the #348 behavior. Not necessarily a problem, but possibly a subtle point of confusion. If you want to switch to the #348 behavior by default when loading existing traditional jobs, you could either
* invert the sense of the flag, so default ~ `false` is the #348-style behavior, as in https://github.com/jenkinsci/junit-plugin/pull/384#discussion_r874738335
* convert to `Boolean` and use `readResolve` to translate `null` to `true`